src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryExtensions.cs

```diff
@@ -36,7 +36,7 @@ public static void AppendCommaSeparatedValues(this IHeaderDictionary headers, st
         /// <returns>the associated values from the collection separated into individual values, or StringValues.Empty if the key is not present.</returns>
         public static string[] GetCommaSeparatedValues(this IHeaderDictionary headers, string key)
         {
-            return ParsingHelpers.GetHeaderSplit(headers, key);
+            return ParsingHelpers.GetHeaderSplit(headers, key).ToArray();
```

Comment:
Though where I do ```.ToString()``` is generally to avoid boxing the param.

Have taken the approach that implicit conversion is nice for users; but in perf sensitive code is better to be explicit about what's actually happening so the costs are more clearly understood when reviewing. Sensible approach?